### ISSUE TITLE
Not getting metrics reported for anything but TTFP and CLS (in chrome), and throwing errors (in safari).

### ISSUE THREAD

**Issue by justinhelmer** [GoogleChrome/web-vitals#116]
I am using the base+polyfill setup, with the polyfill inlined (from https://unpkg.com/web-vitals@1.0.1/dist/polyfill.js). The base (UMD) code is at the bottom of the body. Originally included as a deferred script, it is now also inline (from https://unpkg.com/web-vitals@1.0.1/dist/web-vitals.base.umd.js).

Additionally the DOMContentLoaded event handling at the end of the body, that is just console.logging the metrics.

Since the versions of the polyfill and base library are in-sync, and all of the snippets are placed in the source according to the documentation, I would expect it to "just work". However most of the metrics are not being reported.

In chrome, only TTFB reports a value, and then CLS (only after I leave the tab and come back).

In safari, nothing is reported - instead, an error is thrown: "_window.webVitals.firstInputPolyfill is not a function._"

We are only interested in the Core Web Vitals (FID, CLS, LCP) - but added calls to all of them for the purpose of understanding which metrics work to potentially help deduce the issue.

[Here](https://www.intellimize.com/) is the page in question.

Am I missing something? Any help would be greatly appreciated. Thanks.

**Comment by justinhelmer** [comment]
As a test, I installed the Web Vitals chrome extension, and that is able to report the values for the same URL. However the instrumented code does not.

**Comment by philipwalton** [comment]
Thanks for linking to the site. That was really helpful in tracking down the issue. It looks like the build config for the UMD version was causing it to override the global object set by the polyfill, which was only happening when using the UMD version, so it didn't get caught by the integration tests (which use the module version).

I've [updated this](https://github.com/GoogleChrome/web-vitals/pull/117) and released version `1.1.0`. Let me know if that doesn't fix the issue for you.

**Comment by justinhelmer** [comment]
@philipwalton - I'm very thankful for the quick turnaround.

The new release does in fact fix the issue in chrome, where now all of the web vitals are being reported.

However, in safari, there still appears to be an issue. Although the _"window.webVitals.firstInputPolyfill is not a function."_ error has been resolved, it is now behaving more like chrome was prior to your minor update release. Only FID is being reported, and TTFB is being _intermittently_ reported. No other metrics are being reported.

FWIW, I am on 14.0.2 (16610.3.7.1.9). I have not tested any other browsers.

Thanks again!

**Comment by philipwalton** [comment]
I'm seeing Safari 14.0.2 report TTFB 100% of the time when I visit that page.

One thing to be aware of is TTFB is calculated using the Navigation Timing API, which—since it has timings for all the load-related events up until load—it needs to wait until after the load event fires before it can report TTFB.

Is it possible that some of the times you're not seeing TTFB reported is just that the page hasn't finished loading yet?

**Comment by justinhelmer** [comment]
It is entirely possible that I am seeing an isolated issue related to TTFB; however I am seeing no pending network requests and yet, no TTFB even minutes later. When it works, it shows up within a couple seconds.

Regardless, I will put that concern aside about TTFB, and assume it is somehow isolated to my environment. In reality we only intend to capture the Core Web Vitals (CLS, FID, LCP).

However is it not correct that we should also expect to see values for CLS, LCP, and FCP? All of the values are now being reported in chrome thanks to your fix. However in Safari I am only seeing FID (and sometimes TTFB). FID is great, as that is one of the core web vitals we intend to capture. However missing CLS and LCP means we are not able to get the full picture.

**Comment by philipwalton** [comment]
That is expected. Unfortunately the APIs need to measure these metrics are not available in Safari. It's possible to measure FID with the polyfill, but with the other metrics it's just not possible.

Note that Safari is planning to add FCP support in the near future, so when that happens you'll at least get another metric available to you in Safari.

**Comment by justinhelmer** [comment]
Lovely thanks for clarifying. In that case everything works great!
